src/main/java/org/jabref/logic/citationkeypattern/CitationKeyGenerator.java

```diff
@@ -213,7 +213,7 @@ private String createCitationKeyFromPattern(BibEntry entry) {
             // ":lower":
             if (fieldParts.size() > 1) {
                 // apply modifiers:
-                expandedPattern = applyModifiers(expandedPattern, fieldParts, 1);
+                expandedPattern = applyModifiers(expandedPattern, fieldParts, 1, CitationKeyGenerator.expandBracketContent(keywordDelimiter, entry, database));
```

Comment:
A test case for `CitationKeyGenerator.expandBracketContent` should also be there, shouldn't it?

Reply:
On a side-note, perhaps I miscommunicated. Here, you are still using the method `expandBracketContent` inherited from `BracketedPattern`. I can't really think of a scenario where this would be an issue but it might be better to change it to 
```suggestion
                expandedPattern = applyModifiers(expandedPattern, fieldParts, 1, expandBracketContent(entry));
```